cherab/solps/formats/balance.py

```diff
@@ -0,0 +1,165 @@
+
+Q = elementary_charge
+
+# key is nuclear charge Z and atomic mass AMU
+_popular_species = {
+    (1, 2): deuterium,
+    (6, 12.0): carbon,
+    (2, 4.003): helium,
+    (7, 14.0): nitrogen,
+    (10, 20.180): neon,
+    (18, 39.948): argon,
+    (18, 40.0): argon,
+    (36, 83.798): krypton,
+    (54, 131.293): xenon
+}
+
+
+# Code developed by J. Harrison 9/4/2019
+def load_solps_from_balance(balance_filename):
+    """
+    Load a SOLPS simulation from SOLPS balance.nc output files.
+
+    """
+
+    # Open the file
+    fhandle = netcdf.netcdf_file(balance_filename,'r')
+	
+    # Load SOLPS mesh geometry
+    cr_x = fhandle.variables['crx'].data.copy()
+    cr_z = fhandle.variables['cry'].data.copy()
+    vol = fhandle.variables['vol'].data.copy()
+	
+    # Re-arrange the array dimensions in the way CHERAB expects...
+    cr_x = np.moveaxis(cr_x, 0, -1)
+    cr_z = np.moveaxis(cr_z, 0, -1)
+	
+    # Create the SOLPS mesh
+    mesh = SOLPSMesh(cr_x,cr_z,vol)	
+
+    sim = SOLPSSimulation(mesh)
+    ni = mesh.nx
+    nj = mesh.ny
+
+    # TODO: add code to load SOLPS velocities and magnetic field from files
+
+    # Load electron species
+    sim._electron_temperature = fhandle.variables['te'].data.copy()/Q
+    sim._electron_density = fhandle.variables['ne'].data.copy()
+
+    ##########################################
+    # Load each plasma species in simulation #
+    ##########################################
+
+    sim._species_list = []
+    n_species = len(fhandle.variables['am'].data)
+	
+    for i in range(n_species):
+
+        # Extract the nuclear charge	
+        if fhandle.variables['species'].data[i,1] == b'D':
+            zn = 1
+        if fhandle.variables['species'].data[i,1] == b'C':
+            zn = 6
+        if fhandle.variables['species'].data[i,1] == b'N':
+            zn = 7
+        if fhandle.variables['species'].data[i,1] == b'N' and fhandle.variables['species'].data[i,2] == b'e':
+            zn = 10
+        if fhandle.variables['species'].data[i,1] == b'A' and fhandle.variables['species'].data[i,2] == b'r':
+            zn = 18
+	
+        am = float(fhandle.variables['am'].data[i])  # Atomic mass
+        charge = int(fhandle.variables['za'].data[i])  # Ionisation/charge
+        species = _popular_species[(zn, am)]
+		
+        # If we only need to populate species_list, there is probably a faster way to do this...		
+        sim.species_list.append(species.symbol + str(charge))
+
+    tmp = fhandle.variables['na'].data.copy()
+    tmp = np.moveaxis(tmp, 0, -1)
+    sim._species_density = tmp
+
+    # Make Mesh Interpolator function for inside/outside mesh test.
+    inside_outside_data = np.ones(mesh.num_tris)
+    inside_outside = AxisymmetricMapper(Discrete2DMesh(mesh.vertex_coords, mesh.triangles, inside_outside_data, limit=False))
+    sim._inside_mesh = inside_outside
+
+    
+    # Load the neutrals data
+    D0_indx = sim.species_list.index('D0')
```

Comment:
This will raise a `ValueError` if there is no 'D0' in the species list. Probably safest to wrap this in a try/except block if you want to be able to handle balance.nc files without neutral deuterium:
```python
try:
    D0_indx = sim.species_list.index('D0')
except ValueError:
    D0_indx = None
```
Of course, if missing D0 data is a critical error then there's no point catching this, but then you don't need to do the `if D0_indx is not None` check in line 127.